packages/zoe/src/contracts/attestation/expiringNFT.js

```diff
@@ -0,0 +1,264 @@
+      }
+    });
+
+    assert(
+      foundOldRecord,
+      `No previous lien was found for address ${address} and amount ${amountLiened}`,
+    );
+
+    // commit point
+    lienedAmounts.set(address, minusOldRecord);
+  };
+
+  /**
+   * Look at the amounts liened and their expiration and unlien anything
+   * that is expired.
+   *
+   * @param {Address} address
+   * @param {Timestamp} currentTime
+   * @returns {Amount} amountLiened
+   */
+  const unlienIfExpired = (address, currentTime) => {
+    let totalStillLiened = empty;
+
+    // It is possible that the address does not currently have any
+    // lienedAmounts.
+    if (!lienedAmounts.has(address)) {
+      return empty;
+    }
+
+    const lienedAtStart = lienedAmounts.get(address);
+    const notExpired = lienedAtStart.filter(({ amountLiened, expiration }) => {
+      if (hasExpired(expiration, currentTime)) {
+        return false;
+      }
+      totalStillLiened = AmountMath.add(totalStillLiened, amountLiened);
+      return true;
+    });
+    // commit point
+
+    // Remove from cannotGetExtension since there is no lien left
+    if (AmountMath.isEmpty(totalStillLiened)) {
+      cannotGetExtension.delete(address);
+    }
+
+    lienedAmounts.set(address, notExpired);
+    return totalStillLiened;
+  };
+
+  /** @type {getLienAmount} */
+  const getLienAmount = async (address, currentTime) => {
+    assert.typeof(address, 'string');
+
+    // We should first unlien any amounts for which the lien has
+    // expired, then return the total lien still remaining for the address
+    return unlienIfExpired(address, currentTime);
+  };
+
+  /** @type {ExtendExpiration} */
+  const extendExpiration = (seat, newExpiration) => {
+    const oldAttestationAmount = seat.getAmountAllocated('Attestation');
+
+    const attestationValue =
+      /** @type {SetValue} */ (oldAttestationAmount.value);
+
+    // TODO: allow for multiple elements in the amount value to each
+    // be extended. Currently, we restrict the value to a single element.
+    assert(
+      attestationValue.length === 1,
+      X`We can currently only extend a single attestation element at a time`,
+    );
+    const {
+      expiration,
+      handle,
+      address,
+      amountLiened,
+    } = /** @type {ExpiringAttElem} */ (attestationValue[0]);
+
+    assert(
+      !cannotGetExtension.has(address),
+      `The address ${address} cannot extend the expiration for attestations`,
+    );
+
+    assert(
+      newExpiration > expiration,
+      `The newExpiration ${newExpiration} must be later than the old expiration ${expiration}`,
+    );
+    // Importantly, we cannot drop the uniqueHandle and make a new
+    // one, because that would allow someone to escrow the newly
+    // minted attestation in a contract and the contract would have no
+    // way of knowing it was representing the same value as before,
+    // allowing the user to misrepresent the liened amount as larger
+    // then it actually is.
```

Comment:
```suggestion
    // allowing the user to vote multiple times.
```

Reply:
I don't think that's right. The attestations affect the weight of the voting, not whether the user votes multiple times.